CMakeLists.txt

```diff
@@ -39,6 +44,8 @@ endif (USE_AVX)
 
 # Sanitizer
 if (USE_SANITIZER)
+  # Older CMake versions have had troubles with Sanitizer
```

Comment:
Could you share what's the trouble,  when I introduced sanitizer to XGBoost I was running an older version of CMake.

Reply:
I kept seeing this error when I ran ASAN-enabled testxgboost:
```
==16412==Shadow memory range interleaves with an existing memory mapping. ASan cannot proceed correctly. ABORTING.
==16412==ASan shadow was supposed to be located in the [0x00007fff7000-0x10007fff7fff] range.
==16412==This might be related to ELF_ET_DYN_BASE change in Linux 4.12.
==16412==See https://github.com/google/sanitizers/issues/856 for possible workarounds.
==16412==Process memory map follows:
  0x00007fff7000-0x00008fff7000
  0x00008fff7000-0x02008fff7000
  0x02008fff7000-0x10007fff8000
  0x5577ad855000-0x5577ae3a7000 /workspace/build/testxgboost
  0x5577ae5a6000-0x5577ae5b4000 /workspace/build/testxgboost
  0x5577ae5b4000-0x5577ae61d000 /workspace/build/testxgboost
  0x5577ae61d000-0x5577ae625000
  0x600000000000-0x606000000000
  0x606000000000-0x606000010000
  0x606000010000-0x606e00000000
  0x606e00000000-0x606e00010000
  0x606e00010000-0x607000000000
  0x607000000000-0x607000010000
  0x607000010000-0x607e00000000
  0x607e00000000-0x607e00010000
  0x607e00010000-0x624000000000
  0x624000000000-0x624000010000
  0x624000010000-0x624e00000000
  0x624e00000000-0x624e00010000
  0x624e00010000-0x640000000000
  0x640000000000-0x640000003000
  0x7f4f809b4000-0x7f4f81058000
  0x7f4f81058000-0x7f4f8105f000 /lib/x86_64-linux-gnu/librt-2.27.so
  0x7f4f8105f000-0x7f4f8125e000 /lib/x86_64-linux-gnu/librt-2.27.so
  0x7f4f8125e000-0x7f4f8125f000 /lib/x86_64-linux-gnu/librt-2.27.so
  0x7f4f8125f000-0x7f4f81260000 /lib/x86_64-linux-gnu/librt-2.27.so
  0x7f4f81260000-0x7f4f81263000 /lib/x86_64-linux-gnu/libdl-2.27.so
  0x7f4f81263000-0x7f4f81462000 /lib/x86_64-linux-gnu/libdl-2.27.so
  0x7f4f81462000-0x7f4f81463000 /lib/x86_64-linux-gnu/libdl-2.27.so
  0x7f4f81463000-0x7f4f81464000 /lib/x86_64-linux-gnu/libdl-2.27.so
  0x7f4f81464000-0x7f4f8164b000 /lib/x86_64-linux-gnu/libc-2.27.so
  0x7f4f8164b000-0x7f4f8184b000 /lib/x86_64-linux-gnu/libc-2.27.so
  0x7f4f8184b000-0x7f4f8184f000 /lib/x86_64-linux-gnu/libc-2.27.so
  0x7f4f8184f000-0x7f4f81851000 /lib/x86_64-linux-gnu/libc-2.27.so
  0x7f4f81851000-0x7f4f81855000
  0x7f4f81855000-0x7f4f819f2000 /lib/x86_64-linux-gnu/libm-2.27.so
  0x7f4f819f2000-0x7f4f81bf1000 /lib/x86_64-linux-gnu/libm-2.27.so
  0x7f4f81bf1000-0x7f4f81bf2000 /lib/x86_64-linux-gnu/libm-2.27.so
  0x7f4f81bf2000-0x7f4f81bf3000 /lib/x86_64-linux-gnu/libm-2.27.so
  0x7f4f81bf3000-0x7f4f81c17000 /opt/python/lib/libasan.so.5.0.0
  0x7f4f81c17000-0x7f4f81cca000 /opt/python/lib/libasan.so.5.0.0
  0x7f4f81cca000-0x7f4f81cfd000 /opt/python/lib/libasan.so.5.0.0
  0x7f4f81cfd000-0x7f4f81cfe000 /opt/python/lib/libasan.so.5.0.0
  0x7f4f81cfe000-0x7f4f81d01000 /opt/python/lib/libasan.so.5.0.0
  0x7f4f81d01000-0x7f4f81d04000 /opt/python/lib/libasan.so.5.0.0
  0x7f4f81d04000-0x7f4f8296a000
  0x7f4f8296a000-0x7f4f82984000 /lib/x86_64-linux-gnu/libpthread-2.27.so
  0x7f4f82984000-0x7f4f82b83000 /lib/x86_64-linux-gnu/libpthread-2.27.so
  0x7f4f82b83000-0x7f4f82b84000 /lib/x86_64-linux-gnu/libpthread-2.27.so
  0x7f4f82b84000-0x7f4f82b85000 /lib/x86_64-linux-gnu/libpthread-2.27.so
  0x7f4f82b85000-0x7f4f82b89000
  0x7f4f82b89000-0x7f4f82cd9000 /usr/lib/x86_64-linux-gnu/libasan.so.4.0.0
  0x7f4f82cd9000-0x7f4f82ed9000 /usr/lib/x86_64-linux-gnu/libasan.so.4.0.0
  0x7f4f82ed9000-0x7f4f82edc000 /usr/lib/x86_64-linux-gnu/libasan.so.4.0.0
  0x7f4f82edc000-0x7f4f82edf000 /usr/lib/x86_64-linux-gnu/libasan.so.4.0.0
  0x7f4f82edf000-0x7f4f83b44000
  0x7f4f83b44000-0x7f4f83b6b000 /lib/x86_64-linux-gnu/ld-2.27.so
  0x7f4f83bab000-0x7f4f83bbf000
  0x7f4f83bc3000-0x7f4f83be6000
  0x7f4f83be6000-0x7f4f83be9000 /opt/python/lib/libgcc_s.so.1
  0x7f4f83be9000-0x7f4f83bf6000 /opt/python/lib/libgcc_s.so.1
  0x7f4f83bf6000-0x7f4f83bf9000 /opt/python/lib/libgcc_s.so.1
  0x7f4f83bf9000-0x7f4f83bfa000 /opt/python/lib/libgcc_s.so.1
  0x7f4f83bfa000-0x7f4f83bfb000 /opt/python/lib/libgcc_s.so.1
  0x7f4f83bfb000-0x7f4f83bfd000
  0x7f4f83bfd000-0x7f4f83c91000 /opt/python/lib/libstdc++.so.6.0.25
  0x7f4f83c91000-0x7f4f83cf6000 /opt/python/lib/libstdc++.so.6.0.25
  0x7f4f83cf6000-0x7f4f83d2d000 /opt/python/lib/libstdc++.so.6.0.25
  0x7f4f83d2d000-0x7f4f83d37000 /opt/python/lib/libstdc++.so.6.0.25
  0x7f4f83d37000-0x7f4f83d3b000 /opt/python/lib/libstdc++.so.6.0.25
  0x7f4f83d3b000-0x7f4f83d3e000
  0x7f4f83d3e000-0x7f4f83d47000 /opt/python/lib/libgomp.so.1.0.0
  0x7f4f83d47000-0x7f4f83d5a000 /opt/python/lib/libgomp.so.1.0.0
  0x7f4f83d5a000-0x7f4f83d62000 /opt/python/lib/libgomp.so.1.0.0
  0x7f4f83d62000-0x7f4f83d63000 /opt/python/lib/libgomp.so.1.0.0
  0x7f4f83d63000-0x7f4f83d64000 /opt/python/lib/libgomp.so.1.0.0
  0x7f4f83d64000-0x7f4f83d67000
  0x7f4f83d67000-0x7f4f83d6b000
  0x7f4f83d6b000-0x7f4f83d6c000 /lib/x86_64-linux-gnu/ld-2.27.so
  0x7f4f83d6c000-0x7f4f83d6d000 /lib/x86_64-linux-gnu/ld-2.27.so
  0x7f4f83d6d000-0x7f4f83d6e000
  0x7ffd539c9000-0x7ffd539ea000 [stack]
  0x7ffd539f8000-0x7ffd539fa000 [vvar]
  0x7ffd539fa000-0x7ffd539fc000 [vdso]
  0xffffffffff600000-0xffffffffff601000 [vsyscall]
==16412==End of process memory map
```

Curiously, the error occurs only when an old version of CMake was used. With 3.12, I do not see this error.
I suppose the recent CMake re-factor is somehow interacting badly with ASAN. I don't think I have sufficient bandwidth now to figure out the root cause here; hence I chose the CMake version I know to work.